include/libtorrent/peer_class_set.hpp

```diff
@@ -56,7 +56,7 @@ namespace libtorrent {
 	private:
 
 		// the number of elements used in the m_class array
-		std::uint8_t m_size;
+		std::int8_t m_size;
```

Comment:
this is not obvious I think. since it's only 8 bits to begin with, loosing one is kind of significant